### PR TITLE
[thread-cert] refactor case 9.2.12 and 9.2.13 using pktverify

### PR DESCRIPTION
**Pull request by FiveDimensions**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py [hunk 1]
```diff
@@ -105,6 +112,24 @@ def test(self):
...
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        ED = pv.vars['ED']
+        COMMISSIONER = pv.vars['COMMISSIONER']
+        _ed_pkts = pkts.filter_wpan_src64(ED)
+
+        # Step 3: ED MUST send MGMT_ED_REPORT.ans to the Commissioner and report energy measurements
+        _ed_pkts.filter_coap_request(MGMT_ED_REPORT).must_next()
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py:124]: should check for the destination?

File: tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py [hunk 2]
```diff
@@ -105,6 +112,24 @@ def test(self):
...
+
+        # Step 5: ED MUST send MGMT_ED_REPORT.ans to the Commissioner and report energy measurements
+        _ed_pkts.filter_coap_request(MGMT_ED_REPORT).must_next()
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py:127]: should check for the destination?

File: tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py]: added

File: tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py (hunk 2 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py]: added

File: tests/scripts/thread-cert/Cert_9_2_12_Announce.py [hunk 3]
```diff
@@ -129,14 +120,56 @@ def test(self):
...
+        pkts.filter_wpan_src64(MED).filter_wpan_dst64(LEADER_2).filter_mle_cmd(
+            MLE_CHILD_ID_REQUEST).must_next().must_verify(lambda p: p.wpan.dst_pan == DATASET1_PANID)
+
+        pkts.filter_wpan_src64(MED).filter_ipv6_dst(LINK_LOCAL_ALL_NODES_MULTICAST_ADDRESS).filter_mle_cmd(
+            MLE_ANNOUNCE).must_next().must_verify(
+                lambda p: {CHANNEL_TLV, PAN_ID_TLV, ACTIVE_TIMESTAMP_TLV} == set(p.mle.tlv.type) and p.wpan.dst_pan ==
+                0xffff and p.wpan.aux_sec.key_id_mode == 0x2 and p.wpan.aux_sec.key_source == 0x00000000ffffffff)
+
+        # Step 6: MED MUST respond with an ICMPv6 Echo Reply
+        pkts.filter_ping_reply().filter_ipv6_src_dst(MED_RLOC, LEADER_2_RLOC)
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_12_Announce.py:171]:
```suggestion
        pkts.filter_ping_reply().filter_ipv6_src_dst(MED_RLOC, LEADER_2_RLOC).must_next()
```

File: tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py [hunk 4]
```diff
@@ -42,23 +44,27 @@ class Cert_9_2_13_EnergyScan(thread_cert.TestCase):
...
         },
         ROUTER1: {
+            'name': 'ROUTER',
             'mode': 'rsdn',
             'panid': 0xface,
             'router_selection_jitter': 1,
             'allowlist': [LEADER, ED1]
         },
         ED1: {
+            'name': 'ED',
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py:67]: In this case, the ED DUT can be MED or SED. How do we verify both role within one case?

Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py:67]: There is no special packet to verify for MED/SED in the test plan

Comment (simonlingoogle) [reply tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py:67]: But we need to verify that both MED/SED works in this test case.
The simplest method is to have two almost identical test cases, with different mods for the ED device. But this brings too much redundant code. 

A better way I can think of is to define a base test class and create two subclasses (in two files) one for MED DUT and another for SED DUT. Each subclass define it's own topology with correct modes for SED/MED. 

Thoughts? @FiveDimensions @librasungirl

Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_9_2_13_EnergyScan.py:67]: Currently, there are several cases with different DUT scenarios, implement this as a example to refactor the others

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**